gm4_bat_grenades/data/bat_grenades/functions/main.mcfunction

```diff
@@ -1,2 +1,2 @@
 execute at @e[type=bat] run playsound minecraft:entity.bat.ambient hostile @a[gamemode=!creative,gamemode=!spectator,distance=..7]
-execute as @e[type=bat] at @s if entity @a[gamemode=!creative,gamemode=!spectator,distance=..3] unless entity @s[nbt={NoAI:1b}] run function bat_grenades:explode
+execute as @e[type=bat] at @s if entity @a[gamemode=!creative,gamemode=!spectator,distance=..3] unless entity @s[nbt={NoAI:1b}] unless entity @s[tag=gm4_defused_bat] run function bat_grenades:explode
```

Comment:
You can combine the NoAI check and the defused check in one selector. Put the tag check in front as optimization.

Reply:
I'll put the tag in front, when I combined the check it had to be both NoAI and defused (so it would be NoAI && defused) instead of either NoAI or defused (NoAI || defused). But I may have messed up in my testing,

Reply:
Oh yeah you're right